builder/common/run_config.go

```diff
@@ -847,3 +877,10 @@ func (c *RunConfig) SSMAgentEnabled() bool {
 	hasIamInstanceProfile := c.IamInstanceProfile != "" || c.TemporaryIamInstanceProfilePolicyDocument != nil
 	return c.SSHInterface == "session_manager" && hasIamInstanceProfile
 }
+
+// IsBurstableInstanceType checks if the InstanceType for the config is one
+// of the following types T2, T3a, T3, T4g
+func (c *RunConfig) IsBurstableInstanceType() bool {
```

Comment:
The Terraform provider uses the DesribeInstanceOffering API here to determine if the selected instance is burstable. This is a nice approach for sure. It does introduce a new API call, which I'm not sure if it would introduce a regression as a new set of permissions may be needed. I can follow up on this.